--- ssl/ssl_txt.orig	Sun Jul 17 17:26:59 2016
+++ ssl/ssl_txt.c	Sun Jul 17 17:35:44 2016
@@ -82,6 +82,7 @@
  * OTHERWISE.
  */
 
+#include <inttypes.h>
 #include <stdio.h>
 
 #include <openssl/buffer.h>
@@ -163,7 +164,7 @@
 	}
 
 	if (x->time != 0) {
-		if (BIO_printf(bp, "\n    Start Time: %lld", (long long)x->time) <= 0)
+		if (BIO_printf(bp, "\n    Start Time: %"PRId64, (int64_t)x->time) <= 0)
 			goto err;
 	}
 	if (x->timeout != 0L) {
